src/entities/attendance-option/api/use-create-detail-option.ts

```diff
@@ -33,7 +33,12 @@ export const useCreateDetailOption = ({
       onMutate?.()
     },
     onSuccess: () => {
-      queryClient.invalidateQueries({ queryKey: attendanceOptionQueryKeys.optionDetail(optionKey) })
+      queryClient.invalidateQueries({
+        queryKey: [
+          attendanceOptionQueryKeys.optionList(),
+          attendanceOptionQueryKeys.optionDetail(optionKey),
+        ],
+      })
```

Comment:
[수정 요청]
tanstack-query의 invalidateQueries는 파라미터로 전달받은 쿼리키를 기존의 쿼리들과 비교해서 캐싱이 진행된 쿼리들을 대상으로 stale 상태로 바꾸는 역할을 수행합니다. 작성해주신 코드처럼 invalidateQueries의 쿼리키 파라미터에 서로 다른 쿼리키를 새로운 배열로 묶어 전달할 경우 완전히 다른 쿼리키로 인식하여 optionList와 optionDetail 키를 가진 쿼리들을 올바르게 비교하지 못해 의도하신 validate 하지 못하는 문제가 발생하게 됩니다. 아래와 같이 invalidateQueries를 상황에 맞게 2번 호출하는 것을 추천드리며, 관련 내용 확인 후 해당 세그먼트에 작성해주신 다른 쿼리 로직들도 함께 수정 부탁드리겠습니다.

<br />

```suggestion
      queryClient.invalidateQueries({ queryKey: attendanceOptionQueryKeys.optionList() })
      queryClient.invalidateQueries({ queryKey: attendanceOptionQueryKeys.optionDetail(optionKey) })
```

Reply:
앗 넵 맞습니다 제가 실수했네요🫢 수정하겠습니다.